update version requirements for 2.1

lache-melvin opened the pull request.

jmbrunskill reviewed (approved):
Thanks!
Technically 2.1 will work with an older version, but you won't able able to assign the correct permissions for new features

lache-melvin commented:
> Thanks! Technically 2.1 will work with an older version, but you won't able able to assign the correct permissions for new features

Yeah, I wasn't quite sure what to do with that? Typically those versions are just for breaking API changes... should the required version for GAPS stuff be captured somewhere else?

lache-melvin commented:
Removed the `v7.15.5` requirement from the matrix, as its only needed for GAPS features. Moved to the `Permissions` page 👍 [313f404](https://github.com/msupply-foundation/msupply_docs/pull/49/commits/313f4044e2b1954d5af221978f384c9aac82deeb)
